packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -1199,15 +1200,48 @@ export class ProjectService {
             user_uuid: user.userUuid,
         };
 
-        return this.runQueryAndFormatRows({
+        const explore = await this.getExplore(
             user,
+            projectUuid,
+            exploreName,
+            organizationUuid,
+        );
+        const exploreDimensions = getDimensions(explore);
+
+        const {
+            metricQuery: metricWithOverrideGranularity,
+            oldDimension,
+            newDimension,
+        } = ProjectService.updateMetricQueryGranularity(
             metricQuery,
+            exploreDimensions,
+            dateZoomGranularity,
+        );
+
+        const results = await this.runQueryAndFormatRows({
+            user,
+            metricQuery: metricWithOverrideGranularity,
             projectUuid,
             exploreName,
+            explore,
             csvLimit,
             context: QueryExecutionContext.EXPLORE,
             queryTags,
+            granularity: dateZoomGranularity,
         });
+
+        // TODO quick hack to get the old results on the chart with new granularity
+        // We should investigate if we can make the changes in the frontend
+        // to get the right field instead.
+        const rowsWithGranularity =
+            oldDimension && newDimension
```

Comment:
you might be able to get the `oldDimension` with `metadata.hasADateDimension` which includes all the info of the first date dimension 👍 would that be useful?

Reply:
mm, maybe, but I'm relying on the oldDimension from `updateMetricQueryGranularity` which returns the old and new date dimensions consistently, so I don't think we need to use anything else